src/renderer/eventBus.ts

```diff
@@ -103,7 +104,8 @@ export const dispatchExtEvent = function (event: ExtensionEvent): void {
 /** Add an extension event handler for a specific event type and schedule dispatch. */
 export const addExtEventHandler = function (
   type: eventType,
-  handler: eventHandler
+  handler: eventHandler,
+  state: stateType
 ): void {
   eventHandlers[type] = eventHandlers[type] || [];
   eventHandlers[type].push(handler);
```

Comment:
We're not actually using `state` here. Maybe it's as simple as creating a complex key like this:
```suggestion
  const key = `${type}${state ? `@@${state}` : ''}`;
  eventHandlers[key] = eventHandlers[type] || [];
  eventHandlers[key].push(handler);
```

Old code keeps working as before. New code provides `state` and we use it for the key.

Now `scheduleDispatch` needs to check for new `event.state` property, and recreate the key in the same way.